jgrapht-demo/src/main/java/org/jgrapht/demo/GraphMLExportDemo.java

```diff
@@ -0,0 +1,260 @@
+
+public final class GraphMLExportDemo
+{
+    // Number of vertices
+    private static final int size = 6;
+
+    private static Random generator = new Random(17);
+
+    enum Color
+    {
+        BLACK("black"),
+        WHITE("white");
+
+        private final String value;
+
+        private Color(String value)
+        {
+            this.value = value;
+        }
+
+        public String toString()
+        {
+            return value;
+        }
+
+    }
+
+    static class GraphVertex
+    {
+        private String id;
+        private Color color;
+
+        public GraphVertex(String id)
+        {
+            this(id, null);
+        }
+
+        public GraphVertex(String id, Color color)
+        {
+            this.id = id;
+            this.color = color;
+        }
+
+        @Override
+        public int hashCode()
+        {
+            final int prime = 31;
+            int result = 1;
+            result = prime * result + ((id == null) ? 0 : id.hashCode());
+            return result;
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            if (this == obj)
+                return true;
+            if (obj == null)
+                return false;
+            if (getClass() != obj.getClass())
+                return false;
+            GraphVertex other = (GraphVertex) obj;
+            if (id == null) {
+                if (other.id != null)
+                    return false;
+            } else if (!id.equals(other.id))
+                return false;
+            return true;
+        }
+
+        public Color getColor()
+        {
+            return color;
+        }
+
+        public void setColor(Color color)
+        {
+            this.color = color;
+        }
+
+        public String getId()
+        {
+            return id;
+        }
+
+        public void setId(String id)
+        {
+            this.id = id;
+        }
+
+        @Override
+        public String toString()
+        {
+            return id;
+        }
+    }
+
+    public static void main(String[] args)
+    {
+        /*
+         * Generate complete graph.
+         * 
+         * Vertices have random colors and edges have random edge weights.
+         */
+        DirectedPseudograph<GraphVertex, DefaultWeightedEdge> g = new DirectedPseudograph<>(
+            DefaultWeightedEdge.class);
+        CompleteGraphGenerator<GraphVertex, DefaultWeightedEdge> completeGenerator = new CompleteGraphGenerator<>(
+            size);
+        VertexFactory<GraphVertex> vFactory = new VertexFactory<GraphVertex>()
+        {
+            private int id = 0;
+
+            @Override
+            public GraphVertex createVertex()
+            {
+                GraphVertex v = new GraphVertex(String.valueOf(id++));
+                if (generator.nextBoolean()) {
+                    v.setColor(Color.BLACK);
+                } else {
+                    v.setColor(Color.WHITE);
+                }
```

Comment:
This is fine.  Alternatively you could add a getRandomColor() function to your color enum:
public static Letter getRandomColor() {
        return values()[(int) (Math.random() * values().length)];
    }